lib/index.es6

```diff
@@ -0,0 +1,34 @@
+import path from "path";
+import fs from "fs";
+
+import {sync as mkdirp} from "mkdirp";
+
+import compile from "./compile";
+import {entries} from "./util";
+
+export default function Interlock (options) {
+  const cwd = process.cwd();
+  options = options || {};
+
+  if (!options.emit || !options.emit.length) {
+    throw new Error("Must define at least one bundle.");
+  }
+
+  // TODO: validate options, or only persist specific values
```

Comment:
I've never used Joi but looking at the docs it seems way more involved than validate.js. I can help you evaluate the two but I have no experience personally with either.